agent/php_internal_instrument.c

```diff
@@ -2189,9 +2189,14 @@ NR_INNER_WRAPPER(curl_exec) {
   zval* curlres = NULL;
   int zcaught = 0;
 
-  if (SUCCESS
-      != zend_parse_parameters_ex(ZEND_PARSE_PARAMS_QUIET,
-                                  ZEND_NUM_ARGS() TSRMLS_CC, "r", &curlres)) {
+#if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO /* PHP 8.0+ */
+  int parse_status = zend_parse_parameters_ex(
+      ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS() TSRMLS_CC, "o", &curlres);
+#else
+  int parse_status = zend_parse_parameters_ex(
+      ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS() TSRMLS_CC, "r", &curlres);
+#endif /* PHP 8.0+ */
+  if (SUCCESS != parse_status) {
```

Comment:
Great job on this! I had just a few small comments. 

Minor nit.  

Would it make sense just declare/initialize  
`int parse_status = FAILURE;` before the if/else block?

Similar to here: https://github.com/newrelic/newrelic-php-agent/pull/70/files#diff-d66d558f126dbc42fe98d5332fa13d281fe925d4e01b7608a8dc39c56dce778eR2227

?

Reply:
Perfect, thanks!